test/extensions/filters/http/decompressor/decompressor_filter_integration_test.cc

```diff
@@ -153,4 +155,107 @@ TEST_P(DecompressorIntegrationTest, BidirectionalDecompression) {
       uncompressed_response_length);
 }
 
+/**
+ * Exercises gzip decompression bidirectionally with default configuration.
+ */
+TEST_P(DecompressorIntegrationTest, BidirectionalDecompressionError) {
+  const std::string bad_config{R"EOF(
+      name: default_decompressor
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.filters.http.decompressor.v3.Decompressor
+        decompressor_library:
+          name: testlib
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.compression.gzip.decompressor.v3.Gzip
+            window_bits: 10
+    )EOF"};
+  // Use gzip for decompression.
+  initializeFilter(bad_config);
+
+  // Enable request decompression by setting the Content-Encoding header to gzip.
+  auto encoder_decoder =
+      codec_client_->startRequest(Http::TestRequestHeaderMapImpl{{":method", "POST"},
+                                                                 {":scheme", "http"},
+                                                                 {":path", "/test/long/url"},
+                                                                 {":authority", "host"},
+                                                                 {"content-encoding", "gzip"}});
+  auto request_encoder = &encoder_decoder.first;
+  auto response = std::move(encoder_decoder.second);
+
+  // Send first data chunk upstream.
+  Buffer::OwnedImpl request_data1;
+  TestUtility::feedBufferWithRandomCharacters(request_data1, 8192);
+  request_compressor_->compress(request_data1, Envoy::Compression::Compressor::State::Flush);
+  auto compressed_request_length = request_data1.length();
+  codec_client_->sendData(*request_encoder, request_data1, false);
+
+  // Send second data chunk upstream and finish the request stream.
+  Buffer::OwnedImpl request_data2;
+  TestUtility::feedBufferWithRandomCharacters(request_data2, 16384);
+  request_compressor_->compress(request_data2, Envoy::Compression::Compressor::State::Finish);
+  compressed_request_length += request_data2.length();
+  codec_client_->sendData(*request_encoder, request_data2, true);
+
+  // Wait for frames to arrive upstream.
+  ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
+  ASSERT_TRUE(fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_));
+  ASSERT_TRUE(upstream_request_->waitForEndStream(*dispatcher_));
+
+  EXPECT_TRUE(upstream_request_->complete());
+  TestUtility::headerMapEqualIgnoreOrder(
```

Comment:
Shouldn't we set expectations for these equality checks? Looks like the return value of the check is simply ignored.

Reply:
yeah, they were bad before. Thanks for noticing, updated!